gateway/src/apicast/policy/token_introspection/token_introspection.lua

```diff
@@ -0,0 +1,68 @@
+local policy = require('apicast.policy')
+local _M = policy.new('Token Introspection Policy')
+
+local cjson = require('cjson')
+local http_authorization = require 'resty.http_authorization'
+local http_ng = require 'resty.http_ng'
+local user_agent = require 'apicast.user_agent'
+local resty_env = require('resty.env')
+
+local new = _M.new
+
+function _M.new(config)
+  local self = new()
+  self.config = config or {}
+  self.http_client = http_ng.new{
+    backend = config.client,
+    options = {
+      header = { ['User-Agent'] = user_agent() },
+      ssl = { verify = resty_env.enabled('OPENSSL_VERIFY') }
+    }
+  }
+  return self
+end
+
+local function introspect_token(self, token)
+  local config = self.config
+  if not config then
+    return false
+  end
+
+  local introspection_url = config.introspection_url
+  local client = config.client_id
+  local secret = config.client_secret
+  local credential = 'Basic ' .. ngx.encode_base64(table.concat({ client or '', secret or '' }, ':'))
+  local opts = {
+    headers = {
+      ['Authorization'] = credential
+    }
+  }
+
+  local res, err = self.http_client.post(introspection_url , { token = token, token_type_hint = 'access_token'}, opts)
+  if res and err then
+    ngx.log(ngx.WARN, 'token introspection error: ', err, ' url: ', introspection_url)
+    return false
+  end
+
+  if res.status == 200 then
+    local token_info = cjson.decode(res.body)
+    return token_info.active
```

Comment:
Is this `active` field defined in some RFC? Could we link to the relevant RFC section in a comment?

Reply:
Came by to look at your PR review automation, stayed to help out w/ a little PR doc pointing. :-) Seems the  appropriate RFC doc entry point is: https://tools.ietf.org/html/rfc7662#section-2.2 Where `active` is the only REQUIRED member of the returned JSON introspection result.

Reply:
@reedstrm very nice. Thanks! This looks like the right doc.

@tmogi001 for cases where is RFC available I try to implement it as:
https://github.com/3scale/apicast/blob/5168008db2abb2e217d23f925599fd33da5248cb/gateway/src/resty/http_ng/cache_store.lua#L85-L110

Because the RFC is immutable the comments won't get out of sync and act as relevant context with links to the other parts of the RFC. In this case just the relevant parts to the code in question to document the requests and responses for example.

BTW it is pity[ OpenID Connect Discovery](http://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata) does not define token introspection endpoint. Some IDPs provide it (like Keycloak as `token_introspection_endpoint`), but I don't see it being part of the spec.